qiling/extensions/coverage/formats/history.py

```diff
@@ -81,11 +82,22 @@ def get_ins_exclude_lib(self, libs: list) -> List:
             List: A list of addresses that have been executed and are not in the memory maps that match the regex
 
         Examples:
-            >>> history.get_ins_exclude_lib(["libc.so", "libpthread.so"])
+            >>> history.get_ins_exclude_lib([".*libc.so.*", ".*libpthread.so.*"])
         '''
 
         executable_maps = self.get_regex_matching_exec_maps(libs)
-        return [x for x in self.history if any([x < start or x > end for start, end, _, _, _ in executable_maps])]
+        instructions = []
```

Comment:
The previous version was cleaner and more Pythonic, why did you "expand" it to an explicit loop?

Reply:
```suggestion
return [h for h in self.history if not any(start <= h <= end for start, end, _, _, _ in executable_maps)]
```

Reply:
lol ya i couldn't figure out how to rewrite it to get the test to pass. theres probably a list comprehension way to do it but idk how